benchexec/tablegenerator/react-table/src/utils/utils.js

```diff
@@ -610,6 +623,7 @@ const makeFilterDeserializer =
         out.push({
           id: "id",
           ...tokenizePart(filter),
+          isTableTabFilter: true,
```

Comment:
What does this do? It was added in 2f03808b016150b2c954d27ae86b263eeb39096d, but the commit message only mentions changes to tests.

Reply:
```
const makeFilterSerializer =
  ({ statusValues: allStatusValues, categoryValues: allCategoryValues }) =>
  (filter) => {
    const groupedFilters = {};
    const tableTabIdFilters = [];
    for (const { id, value, values, isTableTabFilter } of filter) {
      if (id === "id") {
        if (values && values.length > 0) {
          groupedFilters.ids = {
            values: values.map((val) => (val ? val : "")),
          };
        } else if (isTableTabFilter) {
          tableTabIdFilters.push({ id, value });
        }
        continue;
      } ...
```
```
if (tableTabIdFilters) {
      tableTabIdFilters.forEach((filter) => {
        runsetFilters.push(
          `id_any(value(${encodeURIComponent(filter.value)}))`,
        );
      });
    }
```
In the `makeFilterSerializer` function, we are pushing to `tableTabIdFilters` array by checking the `isTableTabFilter` property in filter element. Same property should be added while deserializing it but wasn't done. This resulted in id filter getting null even before the recent commits on a special case , i.e if reloaded the page with only id filter applied and then added any other filter.